src/Multi_persistence/include/gudhi/Multi_persistence/Line.h

```diff
@@ -5,253 +5,267 @@
-  U t = inf;
+  /**
+   * @brief Computes the "time" parameter \f$ t \f$ of the starting point \f$ p = base\_point + t \times direction \f$
+   * of the intersection between the line and the open negative cone originating at `x`.
+   *
+   * @tparam U Type of the time parameter.
+   * @param x Origin of the open negative cone.
+   */
+  template <typename U = T>
+  U compute_backward_intersection(const Point &x) const {
+    constexpr const U inf =
+        std::numeric_limits<U>::has_infinity ? std::numeric_limits<U>::infinity() : std::numeric_limits<U>::max();
+    if (x.is_inf()) return inf;
+    if (x.is_minus_inf() || x.is_nan()) return -inf;
+    U t = inf;
 
-  if (direction_.size()) {
-    for (std::size_t i = 0; i < x.size(); i++) {
-      if (direction_[i] == 0) [[unlikely]] {
-        if (x[i] > basepoint_[i])
-          continue;
-        else {
-          return -inf;
+    if (direction_.size()) {
+      for (std::size_t i = 0; i < x.size(); i++) {
+        if (direction_[i] == 0) {
+          if (x[i] <= basePoint_[i]) return -inf;
+        } else {
+          t = std::min(t, (static_cast<U>(x[i]) - static_cast<U>(basePoint_[i])) / static_cast<U>(direction_[i]));
         }
-      } else [[likely]] {
-        t = std::min(t, (static_cast<U>(x[i]) - static_cast<U>(basepoint_[i])) / static_cast<U>(direction_[i]));
       }
+    } else {
+      for (std::size_t i = 0; i < x.size(); i++) t = std::min(t, static_cast<U>(x[i] - basePoint_[i]));
     }
-  } else {
-    for (std::size_t i = 0; i < x.size(); i++) t = std::min(t, static_cast<U>(x[i] - basepoint_[i]));
+    return t;
   }
-  return t;
-}
 
-template <typename T>
+  /**
+   * @brief Computes the "time" parameter \f$ t \f$ of the starting point \f$ p = base\_point + t \times direction \f$
+   * of the intersection between the line and the union of open negative cones originating at the points in `x`.
+   *
+   * @tparam U Type of the time parameter.
+   * @param x Set of origins for the open negative cones.
+   */
+  template <typename U = T>
+  U compute_backward_intersection(const K_critical_point &x) const {
+    constexpr const U inf =
+        std::numeric_limits<U>::has_infinity ? std::numeric_limits<U>::infinity() : std::numeric_limits<U>::max();
+    if (x.is_inf()) return inf;
+    if (x.is_minus_inf() || x.is_nan()) return -inf;
+    U t = -inf;
+    for (const auto &y : x) {
+      t = std::max(t, compute_backward_intersection<U>(y));
+    }
+    return t;
+  }
+
+  /**
+   * @brief Given a box, returns the intersection of this box and the line.
+   *
+   * @param box Box to intersect.
+   * @return A pair representing the two bounding points of the intersection, such that the first element is the
+   * smallest of the two. If the box and the line do not intersect, returns the pair {inf, inf}.
+   */
+  std::pair<Point, Point> get_bounds(const Box<T> &box) const {
+    if (box.is_trivial()) return {Point::inf(), Point::inf()};
```

Comment:
I don't know if {inf, inf} is the best convention, one might genuinely need to consider this box for some reason. We can change that later, if we find a use case for it.

Reply:
What would be the intersection of a trivial box and a line? Or do you mean, you would like to put something else than `{inf, inf}`? Like `{NaN, NaN}` for example?